users/serializers.py

```diff
@@ -267,6 +277,28 @@ def get_unused_coupons(self, instance):
             return fetch_and_serialize_unused_coupons(instance)
         return []
 
+    def validate(self, data):
```

Comment:
Should it be validate_full_name?

Reply:
I have used general validation outside of validate_full_name in order to make name validation required in case if openedx fails and returns an object-level error. We will not allow account creation to proceed and a proper error message will be displayed:
<img width="1280" alt="notification error" src="https://github.com/mitodl/mitxpro/assets/41048311/dc419205-9a27-4167-a080-e0a184b75647">

We have something [similar](https://github.com/mitodl/mitxonline/pull/1389#pullrequestreview-1273522227) suggestion for an object-level error message/handling done in mitxonline. The error message is subjective to what we want to use but I have used it what we were used https://github.com/mitodl/mitxpro/blob/master/authentication/serializers.py#L177 in such cases.

Reply:
Stopping account creation can still be done even we do general validate or not. A general validate would be more useful when we want to throw multiple errors to the frontend at the same time.